### PR TITLE
Issue 2

### LINKED ISSUE #2: Basic Editor based on Codemirror
Create a basic web-based editor using 
* [Codemirror](https://codemirror.net/)
* ~~[LIT](https://lit.dev/)~~ (plain HTML for now)

Supported actions:
* Open / Save UVL file
* Syntax Highlighting + Auto Completion (see #1)

### PR DESCRIPTION
**Pull request by Ateyu**
Feat #2 

The paper is implemented. 
Syntax Highlighting, Error highlighting and quick removal implemented.
Auto completion works for features and operators.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "dependencies": {
+    "@codemirror/autocomplete": "^6.18.0",
+    "@codemirror/highlight": "^0.19.8",
+    "@codemirror/lang-markdown": "^6.2.5",
+    "@codemirror/language": "^6.10.2",
+    "@codemirror/state": "^6.4.1",
+    "@lezer/highlight": "^1.2.1",
+    "@lezer/lr": "^1.4.2",
+    "@rollup/plugin-node-resolve": "^15.2.3",
```
Comment (h3ssto) [inline package.json]: Why are these needed here?

File: src/parser.terms.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// This file was generated by lezer-generator. You probably shouldn't edit it.
+export const
+  FeatureModel = 1,
+  FeatureName = 2,
+  String = 3,
+  FeatureList = 4,
+  Feature3 = 5,
```
Comment (h3ssto) [inline src/parser.terms.js]: ?

**Review by h3ssto — COMMENTED** [review]
Looks alright so far, not sure about the large number of *.mjs.
Features still missing:
* Codemirror
    * [x] Autocompletion
       * [x] Keywords (mandatory, optional, ...)
       * [x] Feature Names (when in constraints)
       * [x] Boolean Operators (when in constraints)
    * [x] Syntax Highlighting
* General
    * [x] Decide on surrounding js framework and wrap

File: src/editor.bundle.js
Comment (h3ssto) [inline src/editor.bundle.js:1]: What is the best practice for bundled files? Probably they should be omitted as they can be always generated?

**Comment by h3ssto** [comment]
Please mark the remaining features and answer the questions above.
Missing in Grammar:
* [x] Force line break after mandatory, optional, or, alternative
* [x] In Cross-Tree Constraints: Add support for & (and) and ! (not) as well as brackets
* [x] Show Syntax Errors (e.g., Space in Feature Names without surrounding " ")

File: src/parser.terms.js (hunk 2 above)
Comment (Ateyu) [reply src/parser.terms.js]: updated and removed. Those were Token from the grammar, that got merged in the newest version

File: src/editor.bundle.js
Comment (Ateyu) [reply src/editor.bundle.js:1]: Best practice is without the bundle

File: package.json (hunk 1 above)
Comment (Ateyu) [reply package.json]: Do you mean why i used scoped packages instead of cdn?

Comment (Ateyu) [reply package.json]: Packages removed. Those were not necessary

File: src/parser.terms.js (hunk 2 above)
Comment (Ateyu) [reply src/parser.terms.js]: parser.tokens.js gets generated to enable indent and dedent but is  removed from repo

**Comment by h3ssto** [comment]
Number of small issues remain:
* [ ] No support for equivalence operator (`<=>`)
* [ ] `sum(Power) < 120 => Large` is flagged as illegal
* [ ] `UndirectedCall => !(FIFO | ShortestPath)` is parsed correctly, but `UndirectedCall => !((((((FIFO))))) | ShortestPath)` is not
* [ ] reserved keywords "features" "constraints" are not enforced currently
* [ ] Is it possible to verify that only existing features are valid in cross-tree constraints?

**Review by h3ssto — APPROVED** [review]
